### PR TITLE
TIP-1079: Fetch group associations by product identifiers

### PR DESCRIPTION
**Pull request by anaelChardan**
**Description (for Contributor and Core Developer)**

Get Groups associated from product identifiers

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetGroupAssociationsByProductIdentifiers.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * It generates product associations with every association types, even if there is not product associated for this association type.
+     * That's why it uses a CROSS JOIN when getting associations at product level.
+     *
+     * @return array ['productA', ['assocType1' => ['groupA']]]
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetGroupAssociationsByProductIdentifiers.php]: arrow + empty assoc (could be done in next PR)

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetGroupAssociationsByProductIdentifiers.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    /** @var Connection */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * It generates product associations with every association types, even if there is not product associated for this association type.
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetGroupAssociationsByProductIdentifiers.php]:
```suggestion
     * It generates product associations with every association types, even if there is no group associated for this association type.
```

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetGroupAssociationsByProductIdentifiers.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    /** @var Connection */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * It generates product associations with every association types, even if there is no group associated for this association type.
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetGroupAssociationsByProductIdentifiers.php]:
```suggestion
     * It generates product associations with every association type, even if there is no associated group for this association type.
```

Comment (mmetayer) [reply src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetGroupAssociationsByProductIdentifiers.php]: :trollface:

File: tests/back/Pim/Enrichment/Integration/Product/Query/Sql/GetGroupAssociationsByProductIdentifiersIntegration.php [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+    public function testWithAProductContainingNoAssociation()
+    {
+        $expected = ['productA' => $this->getAssociationsFormattedAfterFetch()];
+        $actual = $this->getQuery()->fetchByProductIdentifier(['productA']);
+
+        $this->assertEqualsCanonicalizing($expected, $actual);
+    }
+
+    public function testOnASingleProduct()
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Integration/Product/Query/Sql/GetGroupAssociationsByProductIdentifiersIntegration.php]: As seen IRL, I don't really see the point of this test, I would have preferred a test with an empty array as argument

File: tests/back/Pim/Enrichment/Integration/Product/Query/Sql/GetGroupAssociationsByProductIdentifiersIntegration.php [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+    {
+        $expected = ['productC' => $this->getAssociationsFormattedAfterFetch(['groupA', 'groupB'], ['groupC'])];
+        $actual = $this->getQuery()->fetchByProductIdentifier(['productC']);
+
+        $this->assertEqualsCanonicalizing($expected, $actual);
+    }
+
+    public function testOnMultipleSimpleProduct()
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Integration/Product/Query/Sql/GetGroupAssociationsByProductIdentifiersIntegration.php]:
```suggestion
    public function testOnMultipleSimpleProducts()
```

**Pull request merged**